index.html

```diff
@@ -6883,6 +7140,303 @@ <h2 id="MediaStreamAudioSourceNode">
+          example, throwing an exception in case of invalid parameters), and
+          some part happens on the <a>rendering thread</a> (for example,
+          changing the value of an <a>AudioParam</a>).
+        </p>
+        <p>
+          In the description of each operation on <a>AudioNode</a>s and
+          <a>AudioContext</a>s, the synchronous section is marked with a ⌛. All
+          the other operations are executed <a href=
+          "https://html.spec.whatwg.org/multipage/infrastructure.html#in-parallel">
+          in parallel</a>, as described in [[HTML]].
+        </p>
+        <p>
+          The synchronous section is executed on the <a>control thread</a>, and
+          happens immediately. If it fails, the method execution is aborted,
+          possibly throwing an exception. If it succeeds, a <a>control
+          message</a>, encoding the operation to be executed on the
+          <a>rendering thread</a> is enqueued on the <a>control message
+          queue</a> of this <a>rendering thread</a>.
+        </p>
+        <p>
+          The synchronous and asynchronous sections order with respect to other
+          events MUST be the same: given two operation <em>A</em> and
+          <em>B</em> with respectively synchronous and asynchronous section
+          <em>A<sub>Sync</sub></em> and <em>A<sub>Async</sub></em>, and
+          <em>B<sub>Sync</sub></em> and <em>B<sub>Async</sub></em>, if
+          <em>A</em> happens before <em>B</em>, then <em>A<sub>Sync</sub></em>
+          happens before <em>B<sub>Sync</sub></em>, and
+          <em>A<sub>Async</sub></em> happens before <em>B<sub>Async</sub></em>.
+        </p>
+      </section>
+      <section id="rendering-loop">
+        <h3>
+          Rendering an audio graph
+        </h3>
+        <p>
+        Rendering an audio graph is done by blocks of 128 samples-frames.
+        </p>
+
+        <p>
+        Operations that happens <dfn data-lt="atomic">atomically</dfn> on a given thread, can only
+        be executed when no other <a>atomic</a> operation is
+        running on another thread.
+        </p>
+
+        <p>
+        The algorithm for rendering a block of audio from an <a>AudioContext</a>
+        <em>G</em> with an <a href=#control-message-queue>control message queue</a>
+        <em>Q</em> is as follow:
+        </p>
+        <ol>
+          <li> Let <em>Q<sub>rendering</sub></em> be an empty <a
+            href=#control-message-queue>control message queue</a>.
+          <a>Atomically</a> swap <em>Q<sub>rendering</sub></em> and
+          <em>Q</em></li>
+        <li> Until there is no more event in <em>Q<sub>rendering</sub></em>,
+        execute the following steps:
+        <ol>
+          <li>
+          Execute the asynchronous section of the oldest element of
+          <em>Q<sub>rendering</sub></em>.
+          </li>
+          <li>
+          Remove the oldest element of <em>Q<sub>rendering</sub></em>.
```

Comment:
I don't understand the repeated iterations through Qrendering and the nested "Until..." loops in this area. It seems as though there may be some sort of copy/paste error that has repeated some material? Confusingly, the first loop refers to _elements_ of the queue and the next loop refers to _messages_ in the queue... which makes it seem like there is a mixup somewhere here.